File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.stream.*;
import java.util.*;
class Solution {
    public int[] solution(int n, String[] words) {
        List<List<String>> ll = new ArrayList<>();
        
        for(int i = 0; i < n ; i++){
            ll.add(new ArrayList<String>());
        }
    
        Stack<String> preWords = new Stack<>();

        // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다. 
        // System.out.println(ll[0]);
        
        
        for(int i = 0; i< words.length ; i++){
            int nowNum = i % n;
            
            boolean isPre = preWords.contains(words[i]);
            
            if(preWords.isEmpty()){
                ll.get(nowNum).add(words[i]);
                preWords.add(words[i]);
                continue;
            }
                
            
            if(
                isPre || 
                preWords.peek().charAt(preWords.peek().length() - 1) != words[i].charAt(0)
            ){
                return new int[]{nowNum + 1, ll.get(nowNum).size() + 1};
            }
            
            ll.get(nowNum).add(words[i]);
            preWords.add(words[i]);
        }

        return new int[]{0,0};
    }
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let map = {[words[0]] : 0}
    let result;
    for(let i = 1; i < words.length; i++){
        if (words[i - 1].at(-1) !== words[i][0] || words[i] in map) {
            result = [i % n + 1, Math.floor(i / n) + 1];
            break;
        }
        
        map[words[i]] = 0;
    }
    
    return result ?? [0, 0]
}